provider/resource.go

```diff
@@ -23,52 +27,95 @@ func AwsApiGatewayResource() *schema.Resource {
 		DeleteContext: resourceDelete,
 
 		Schema: map[string]*schema.Schema{
-			"api_gateways": {
-				Type:     schema.TypeList,
-				Required: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
-			},
-			"action": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				Default:      string(INCLUDE),
-				ValidateFunc: validation.StringInSlice(ApiGatewayActions, false),
-			},
-			"identifier": {
+			keys.Identifier: {
 				Type:     schema.TypeString,
 				Optional: true,
 				Default:  "",
 			},
-			"ignore_access_log_settings": {
+			keys.IgnoreAccessLogSettings: {
 				Type:     schema.TypeBool,
 				Optional: true,
 				Default:  false,
 			},
-			"log_group_names": {
+			keys.LogGroupNames: {
 				Type:     schema.TypeList,
 				Computed: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+			keys.Accounts: {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						keys.Region: {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						keys.ApiList: {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						keys.CrossAccountRoleArn: {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						keys.Exclude: {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
 func resourceCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	apiGateways := d.Get("api_gateways").([]interface{})
-	action := d.Get("action").(string)
-	ignoreAccessLogSettings := d.Get("ignore_access_log_settings").(bool)
+	logGroupNames := make([]string, 0)
+	accounts := d.Get(keys.Accounts).([]interface{})
+
 	mapDiagnostics := &MapDiagnostics{
 		diagnostics:      diag.Diagnostics{},
 		warnDiagnostics:  make(map[string][]string),
 		errorDiagnostics: make(map[string][]string),
 	}
-	logGroupNames := getLogGroupNames(apiGateways, action == string(EXCLUDE), ignoreAccessLogSettings, meta, mapDiagnostics)
+
+	conn := meta.(AwsApiGatewayProvider)
+
+	for _, account := range accounts {
+		acc := account.(map[string]interface{})
```

Comment:
There's no chance that these type casts would error, right?

Reply:
No, if there's a type mismatch, then terraform will throw an error before calling into the provider as its strictly typed there